File: mwutil/modules/setup-gerrit.py
import os
from argparse import ArgumentParser, Namespace

from mwutil.module import MWUtilModule
from mwutil.utils import MWUtilConfig, run_command, set_git_config


class SetupGerrit(MWUtilModule):

    def get_description(self) -> str:
        return "Sets up a local repository that was cloned from gerrit"

    def populate_subparser(self, parser: ArgumentParser, basedir: MWUtilConfig):
        pass

    def execute(self, config: MWUtilConfig, args: Namespace):
        git_email = os.getenv("GIT_EMAIL")
        git_username = os.getenv("GIT_USERNAME")
        gerrit_username = os.getenv("GERRIT_USERNAME")

        set_git_config("user.email", git_email)
        set_git_config("user.name", git_username)
        set_git_config(f"url.\"ssh://{gerrit_username}@gerrit.wikimedia.org:29418/\".insteadOf", "\"https://gerrit.wikimedia.org/r/\"")
        set_git_config("gitreview.username", git_username)
        set_git_config("gitreview.remote", "origin")

        run_command(["git", "review", "-s", "--verbose"])

        print("Done!")

File: rs/src/modules/setup_gerrit.rs
use crate::config::MWUtilConfig;
use crate::utils::set_git_config;
use anyhow::anyhow;
use std::env;
use std::path::PathBuf;
use std::process::Command;

pub fn execute(config: &MWUtilConfig, repo_folder: Option<PathBuf>) -> anyhow::Result<()> {
    let repo_folder = repo_folder.unwrap_or(env::current_dir()?);
    set_git_config(
        "user.email",
        config.git_email.clone().ok_or_else(|| anyhow!("Git email not set!"))?.as_str(),
        &repo_folder
    )?;
    set_git_config(
        "user.name",
        config.git_username.clone().ok_or_else(|| anyhow!("Git name not set!"))?.as_str(),
        &repo_folder
    )?;
    set_git_config(
        &format!(
            "url.\"ssh://{}@gerrit.wikimedia.org:29418/\".insteadOf",
            config.gerrit_username.clone().ok_or_else(|| anyhow!("Gerrit username not set!"))?.as_str()
        ),
        "\"https://gerrit.wikimedia.org/r/\"",
        &repo_folder
    )?;
    set_git_config(
        "gitreview.username",
        config.git_username.clone().ok_or_else(|| anyhow!("git-review username not set!"))?.as_str(),
        &repo_folder
    )?;
    set_git_config("gitreview.remote", "origin", &repo_folder)?;
    Command::new("git")
        .args(["review", "-s", "--verbose"])
        .current_dir(repo_folder)
        .status()?;
    Ok(())
}
